spec/controllers/filters_controller_spec.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+# rubocop:disable Metrics/BlockLength
+RSpec.describe FiltersController, type: :controller do
+  describe 'GET #show' do
+    before { request.headers['X-RH-IDENTITY'] = encoded_header }
+    let(:endpoint) { FactoryBot.create(:http_endpoint, :account => account) }
+    let(:filter) { endpoint.create_filter(:account => endpoint.account) }
+
+    it 'returns 404 if the endpoint does not exist' do
+      get :show, params: { endpoint_id: endpoint.id + 1 }
+      expect(response).to have_http_status(:not_found)
+      error = JSON.parse(response.body)['errors']
+      expect(error).to eq("Could not find Endpoint with 'id'=#{endpoint.id + 1}")
+    end
+
+    it 'succeeds if the endpoint exists' do
+      filter
+      get :show, params: { endpoint_id: endpoint.id }
+      expect(response).to have_http_status(:ok)
+      data = JSON.parse(response.body)['data']
+      expect(data['id']).to eq(filter.id.to_s)
+      expect(data['attributes']['enabled']).to be_truthy
+      relationships = data['relationships']
+      expect(relationships['apps']['data']).to eq([])
+      expect(relationships['event_types']['data']).to eq([])
+      expect(relationships['levels']['data']).to eq([])
+      endpoint_data = relationships['endpoint']['data']
+      expect(endpoint_data['id']).to eq(endpoint.id.to_s)
+      expect(endpoint_data['type']).to eq('endpoint')
+    end
+
+    it 'succeeds if the endpoint exists but does not have an endpoint' do
```

Comment:
Do you actually need it to fail silently? Why not returning `404`, it's more restful way?

Reply:
The frontend was expecting that. I'll open a PR to the frontend repo as well